src/api/courses.js

```diff
@@ -69,6 +69,124 @@ router.get(
   })
 )
 
+// Get course queue data
+router.get(
+  '/:courseId/data',
+  [requireCourseStaff, requireCourse, failIfErrors],
+  safeAsync(async (req, res, _next) => {
+    const { id: courseId } = res.locals.course
+    const questions = await Question.findAll({
+      include: [
+        {
+          model: Queue,
+          include: [
+            {
+              model: Course,
+              attributes: [['name', 'CourseName']],
+              required: true,
+              where: { courseId: Sequelize.col('queue.courseId') },
+            },
+          ],
+          attributes: [
+            ['id', 'queueId'],
+            'courseId',
+            ['name', 'QueueName'],
+            ['location', 'QueueLocation'],
+            [
+              Sequelize.fn(
+                'datetime',
+                Sequelize.col('queue.createdAt'),
+                'localtime'
+              ),
+              'Queue_CreatedAt',
+            ],
+          ],
+          required: true,
+          where: { courseId, id: Sequelize.col('question.queueId') },
+        },
+        {
+          model: User,
+          as: 'askedBy',
+          attributes: {
+            include: [
+              ['netid', 'AskedBy_netid'],
+              ['universityName', 'AskedBy_RealName'],
+            ],
+            exclude: [
+              'createdAt',
```

Comment:
Is excluding all of these actually required? This seems somewhat brittle if we were to add more fields to users in the future; ditto for other models here.

Reply:
There seemed to be some weirdness with the 'name' field.  If I included just netid and universityName, additional fields called `universityName` and `preferredName` would show up in the response.

Reply:
I can add a test to make sure if fields are added, that it fails if they are not excluded.